### PR TITLE
[CPDNPQ-880] Add api/v1/npq/users#update and api/v1/npq/users#show API endpoints

### PR DESCRIPTION
**Pull request by Lockyy**
### Context

- Ticket: https://dfedigital.atlassian.net/browse/CPDNPQ-880

To allow for historical syncing of UIDs from the Get an Identity service from the NPQ application into the ECF application database we need to implement an API that can accept a new `get_an_identity_id` for a given User.

Alongside that to check data we need a lookup API for checking data.

### Changes proposed in this pull request

- The ECF API supports a new update API endpoint at `PATCH /api/v1/npq/users/:user_id`
- The API accepts the `get_an_identity_id` param which when sent sets that on the user
- The API requires you to use an NPQ specific API key
- If a User already has a `get_an_identity_id` set, the API returns an error and makes no change

- The ECF API supports a new show API endpoint for users at `GET /api/v1/npq/users/:user_id`

#### Response

```
{
  "data" => {
    "id" => user.id.to_s,
    "type" => "user",
    "attributes" => {
      "email" => …,
      "full_name" => …,
      "get_an_identity_id" => …,
    },
  },
}
```

### CODE REVIEW

**Review by shjohnson — APPROVED** [review]

**Comment by shjohnson** [comment]
Couple comments but LGTM!

File: app/controllers/api/v1/npq/users_controller.rb [hunk 1]
```diff
@@ -4,35 +4,73 @@ module Api
...
+
+          if @user.present?
+            render_user(user: @user, status: :ok)
+          else
+            hash = {
+              status: "404",
+              error: "User not found",
+            }
+
+            render json: hash, status: :not_found
```
[the remark this answers is not in the document]
Comment (Lockyy) [reply app/controllers/api/v1/npq/users_controller.rb:18]: Hm, possibly not. Let me check other API responses we have on other endpoints. I'd like to match other APIs we already have.

[the remark this answers is not in the document]
Comment (Lockyy) [reply app/controllers/api/v1/npq/users_controller.rb:18]: Good catch, 401s don't include the status code, I'll update to remove this. 🙂

File: spec/requests/api/v1/npq/users/show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    let(:url) { "/api/v1/npq/users/#{user_id}.json" }
+
+    let(:user) { create(:user, get_an_identity_id: SecureRandom.uuid) }
+    let(:user_id) { user.id }
+
+    def send_request
+      get url, headers: { "Content-Type" => "application/json" }
+    end
+
+    def slice_relevant_user_data(user)
```
[the remark this answers is not in the document]
Comment (Lockyy) [reply spec/requests/api/v1/npq/users/show_spec.rb]: These weren't actually even used in the show and update specs, I've removed them both.

[the remark this answers is not in the document]
Comment (shjohnson) [reply spec/requests/api/v1/npq/users/show_spec.rb]: ahh cool! missed this as well but saw it used other places

File: app/controllers/api/v1/npq/users_controller.rb [hunk 3]
```diff
@@ -4,35 +4,68 @@ module Api
...
+          serialized_user = Api::V1::NPQ::UserSerializer.new(user).serializable_hash
+
+          render json: serialized_user, status:
+        end
+
+        def render_errors(errors:)
+          hash = {
+            errors: errors.map do |error|
+              {
+                status: "400",
```
Comment (shjohnson) [inline app/controllers/api/v1/npq/users_controller.rb:52]: Sorry i actually meant where is this used if its declared below right?

Comment (shjohnson) [inline app/controllers/api/v1/npq/users_controller.rb:52]:
```suggestion
```

**Review by shjohnson — APPROVED** [review]

**Pull request merged**